### ISSUE TITLE
[PRE REVIEW]: catwoman: A transit modelling Python package for asymmetric light curves

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2330]
**Submitting author:** @KathrynJones1 (<a href="http://orcid.org/0000-0002-2316-6850">Kathryn Jones</a>)
**Repository:** <a href="https://github.com/KathrynJones1/catwoman" target ="_blank">https://github.com/KathrynJones1/catwoman</a>
**Version:** v1.0.4
**Editor:** @xuanxu
**Reviewers:** @christinahedges, @barentsen
**Managing EiC:** Lorena A Barba

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**Author instructions**

Thanks for submitting your paper to JOSS @KathrynJones1. **Currently, there isn't an JOSS editor assigned** to your paper.

@KathrynJones1 if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1086/683602 is OK
- 10.1086/345520 is OK
- 10.3847/1538-3881/aae8e5 is OK
- 10.3847/1538-4357/ab55d9 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02330/joss.02330/10.21105.joss.02330.pdf)

**Comment by labarba** [comment]
Is this JOSS submission to accompany a paper to be published with AAS journals? https://joss.readthedocs.io/en/latest/editing.html#handling-of-papers-published-together-with-aas-publishing

**Comment by KathrynJones1** [comment]
Yes we are going to submit a paper to AJ as well

**Comment by arfon** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.63 s (134.7 files/s, 65402.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript                      22           3484           2741          13021
CSS                             14           1133            160          11054
Sass                             7            249            278           2516
SVG                              6              0              0           2405
HTML                             6             51              0            755
C                                8            220            259            690
Python                           7            172            321            567
C/C++ Header                     1             98             32            441
reStructuredText                 7            132             42            226
TeX                              1              8              0             72
Markdown                         1             13              0             40
DOS Batch                        1              8              1             26
Ruby                             1              5              4             11
make                             1              4              6              9
Bourne Shell                     2              0              0              8
-------------------------------------------------------------------------------
SUM:                            85           5577           3844          31841
-------------------------------------------------------------------------------


Statistical information for the repository '2330' was gathered on 2020/06/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Brett M. Morris                  1             1              1            0.01
Brett Morris                     3            86              8            0.24
Ethan Kruse                      5            87             84            0.45
Karl Fogel                       1           215              0            0.56
Kathryn Jones                  115         23230           4244           71.51
Laura Kreidberg                137          6210           3221           24.55
Michael Zhang                   11           397            603            2.60
Nestor Espinoza                  3             7              9            0.04
Simon Walker                     2             5              3            0.02
iancrossfield                    1             3              4            0.02

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Brett Morris                 26           30.2         51.5                7.69
Karl Fogel                   79           36.7         57.7               93.67
Kathryn Jones             20571           88.6          7.1               14.41
Laura Kreidberg            1149           18.5         53.5               18.10
Michael Zhang               235           59.2         36.1               14.04
Simon Walker                  4           80.0         41.0                0.00
```

**Comment by arfon** [comment]
/ cc @crawfordsm for visibility from AAS.

**Comment by arfon** [comment]
:wave: @KathrynJones1, thanks for your submission to JOSS. Before we find an editor to handle this submission I wanted to ask a few questions.

This looks to be an evolution of the [`batman`](https://github.com/lkreidberg/batman) project, with the changes specific to `catwoman` here: https://github.com/lkreidberg/batman/compare/master...KathrynJones1:master . Could you say a little about the longer-term plans for this package? Is `batman` essentially deprecated now (I note there were a couple of small changes to `batman` recently but development mostly seems to have ceased).

The JOSS review process is going to inevitably include reviews a components that were developed originally for `batman` so if there's any long-term plan to merge these two codebases then reviewing `carwoman` in this way might not be optimal.

Relatedly, given the history of the codebase, it would seem appropriate to include the major contributors to `batman` as authors. Is there a reason not to include them?

**Comment by KathrynJones1** [comment]
tagging @nespinoza as co-author

**Comment by nespinoza** [comment]
Hi @arfon, I can chime in here as we were worried by your last point even before starting this project.

To answer your first question: no, `batman` is not deprecated. People still use it, and Laura Kreidberg (the lead developer) answers to questions and comments about the code to this day. It is heavily used in the exoplanet community in fact, and it will keep being this way for a while I believe. If anything, the lack of commits in the last months/years in `batman` speaks of the good coding behind it, as no major bugs have been discovered in the code for a large time-span!

As for your second question: within `catwoman`, we wanted to use some functions in the `batman` C code but these were not easily importable. This is why we decided to write the `catwoman` algorithm by modifying the core `batman` C code in order to (a) not re-write code that is already out there, (b) give proper credit to Laura for those functions (as it is written in each of the scripts in `catwoman`) and (c) use the same code-structure and python class management that `batman` uses, so it was easier for users to use `catwoman` (as most folks in the exoplanet community are familiar with `batman`). 

Before moving forward with that plan, we had a meeting with Laura in order to share it with her and to see if she was uneasy with anything we had in mind. We all agreed that `catwoman` had to be an independent endeavor to that of `batman`, because they attack different geometrical problems, despite the fact that we were going to build this taking `batman` as a base, and also because being `catwoman` a new code only tested and validated by us, it made sense that we took care of any issues that could come our way in terms of bugs, versioning, etc. We also agreed that we were going to reference `batman` both in the paper and in the actual scripts (as we do! --- the latter is actually imposed by the `batman` LICENSE, and we made sure to follow that at the finest detail). We briefly discussed authorship and decided that it would make sense to add her only if she had a direct involvement in the development of `catwoman`, which was not the case. 

Given the above, I think the author list as it is right now is OK. If this makes JOSS, or you @arfon, or @lkreidberg herself uneasy at this stage, we would of course be willing to hear recommendations on this side of things. We strive for properly acknowledging the effort put on `batman`, which is on what `catwoman` was build upon.

**Comment by lkreidberg** [comment]
Thanks for checking on this! It's totally fine with me to proceed with the JOSS paper as written. I didn't contribute at all to the new additions in `catwoman`, so I don't feel any need to be included as a coauthor - an acknowledgement would be fine. 

Congrats @KathrynJones1 and @nespinoza for getting this written up!

**Comment by arfon** [comment]
:wave: @xuanxu - would you be able to edit this submission?

**Comment by arfon** [comment]
Thanks for the explanation @nespinoza and for checking in here @nespinoza, this is helpful.

I still have some concerns about the approach being taken here, specifically a general question about what this fork means for the long term of both of these packages. Our of our [submission criteria](https://joss.readthedocs.io/en/latest/submitting.html) is designed to guard against one-off modifications of projects:

> _...and designed for maintainable extension (not one-off modifications)._

but I'm not sure this applies quite here as the modification is actually describing a different transit geometry problem (and I understand why this means it can't easily be part of `batman`).

I've invited @xuanxu to edit this and I think it would be useful for him to give his opinion before deciding for sure whether we should review this.

Thanks for your patience all!

**Comment by nespinoza** [comment]
Hi @arfon --- indeed, it is not a one-off modification. The fork in our minds just meant making sure to always let people know that we used `batman` as a base to build `catwoman`. The alternative would have been to just download the `batman` repo, do the modifications, and push that to the `catwoman` repo, which didn't seem right to us, but perhaps would have created less confusion.

In essence, right now we think of `catwoman` as a spin-off of `batman`, in that they are now two different libraries supported by independent groups. We are happy to receive any feedback on how to make all this clearer and concern-free :-).

**Comment by xuanxu** [comment]
I see `catwoman` as a different software than `batman`, independently maintained and with no future merge plans, so I think it meets the JOSS submission criteria.

I'd also add it would be nice if the problems that made `catwoman` a fork of `batman` are addressed: so if someone wants to make use of some `catwoman` functions, it would be great if they are easily importable, using Python modules or similar to make something like this work on other people's code:
```python
import catwoman.eclipse
```

**Comment by xuanxu** [comment]
@whedon assign me as editor

**Comment by whedon** [comment]
OK, the editor is @xuanxu

**Comment by xuanxu** [comment]
@jessie-dotson would you be willing to review this paper? I think your knowledge on exoplanets and Python makes you a great fit for this submission.

**Comment by xuanxu** [comment]
@eschnett would you be willing to review this paper? I think your knowledge of astrophysics and C/Python makes you a great fit for this submission.

**Comment by eschnett** [comment]
@xuanxu This paper is outside my expertise – I know next to nothing about exoplanet discovery and the methods and algorithms used there.

**Comment by jessie-dotson** [comment]
Sure. I’d be happy to.

________________________________
On: 23 June 2020 05:56, "Juanjo Bazán" <notifications@github.com<mailto:notifications@github.com>> wrote:

@jessie-dotson<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jessie-2Ddotson&d=DwMCaQ&c=ApwzowJNAKKw3xye91w7BE1XMRKi2LN9kiMk5Csz9Zk&r=EsjtQGK5vNxVnulnNG39CtrT0PFIQKgMF4MiHg-NSFI&m=_73QTl0xvpO5aQe21GhuZn8-6JZJP2EHCg-VLJ8UOE4&s=dq-Wggi27WiSbYo9-RxUL4teSClncMhARAy5CzizHew&e=> would you be willing to review this paper? I think your knowledge on exoplanets and Python makes you a great fit for this submission.

**Comment by xuanxu** [comment]
@jessie-dotson: Great, thanks! Once I have the second reviewer I'll start a new issue where the actual review will take place. Stay tuned!

**Comment by xuanxu** [comment]
@christinahedges: would you be willing to review this paper? Your expertise with exoplanets and Python would come in handy.

**Comment by christinahedges** [comment]
Hi @xuanxu I'm not able to review this at the moment, but please keep me in mind for the next paper!

**Comment by xuanxu** [comment]
@barentsen: would you be willing to review this paper? I think your expertise with exoplanets and Python makes you a great fit for this submission.

**Comment by barentsen** [comment]
@xuanxu Happy to help! :+1:

**Comment by xuanxu** [comment]
@barentsen Great, thanks!
I'll assign you now and start the review issue.

**Comment by xuanxu** [comment]
@whedon add @christinahedges as reviewer

**Comment by whedon** [comment]
OK, @christinahedges is now a reviewer

**Comment by xuanxu** [comment]
@whedon add @barentsen as reviewer

**Comment by whedon** [comment]
OK, @barentsen is now a reviewer

**Comment by xuanxu** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/2382.

**Comment by xuanxu** [comment]
@christinahedges @barentsen please head over to #2382 where the review will take place.

**Comment by christinahedges** [comment]
@xuanxu I am not currently able to review, please see my comment above. I think you mean @jessie-dotson

**Comment by xuanxu** [comment]
@christinahedges you are totally right, sorry!

**Comment by xuanxu** [comment]
@whedon assign @jessie-dotson as reviewer

**Comment by whedon** [comment]
OK, @jessie-dotson is now a reviewer

**Comment by xuanxu** [comment]
@whedon add @barentsen as reviewer

**Comment by whedon** [comment]
OK, @barentsen is now a reviewer

**Comment by xuanxu** [comment]
@jessie-dotson please head over to #2382 where the review will take place.
